src/extensions/FileDropExtension/FileDropExtension.ts

```diff
@@ -0,0 +1,55 @@
+import { Instance } from '@src/Instance';
+import { Module } from '@src/Module';
+import { HTML5Player } from '@src/player/HTML5Player/HTML5Player';
+import { Events } from '@src/types';
+
+export class FileDropExtension extends Module {
+  public name: string = 'FileDropExtension';
+  private dragOverClassName: string = 'ig-drag-over';
+
+  constructor(instance: Instance) {
+    super(instance);
+
+    this.instance.container.ondragover = this.onDragOver;
+    this.instance.container.ondragleave = this.removeDragOverClassName;
+    this.instance.container.ondrop = this.onDrop;
+  }
+
+  private removeDragOverClassName = (e?: DragEvent) => {
+    if (e) {
+      e.preventDefault();
+    }
+
+    this.instance.container.classList.remove(this.dragOverClassName);
+  }
+
+  private onDragOver = (e: DragEvent) => {
+    e.preventDefault();
+    this.instance.container.classList.add(this.dragOverClassName);
+  }
+
+  private onDrop = (e: DragEvent) => {
+    e.preventDefault();
+    this.removeDragOverClassName();
+
+    const files: File[] = Array.from(e.dataTransfer.files);
+    const videoFile = files.find(file => file.type.startsWith('video/'));
+
+    if (videoFile) {
+      this.loadVideo(videoFile);
+    }
+  }
+
+  private loadVideo = (videoFile: File) => {
+    const player = this.instance.player as HTML5Player;
+
+    if (!player.mediaElement.paused) {
+      this.instance.once(Events.STATE_DURATION_CHANGE, () => {
+        this.instance.play();
```

Comment:
I don't like this `once` statement. Is there a better way of checking when the player has loaded the new file set by `player.setSource`?